### DISCUSSION TITLE
Rethink discrete_alloc operator arguments

### DISCUSSION THREAD

**Discussion by jipclaassens** [ObjectVision/GeoDMS#273] [Ideas]
The discrete_alloc operator now needs a lot of arguments, which might not all be as relevant anymore?

Can we omit the feasible solution? 
And the partitioning? (which actually requires an explicitly configured attribute called id in the AggRegio unit). 
And the partitioning name

And more leniency with respect to the value types?
And the claims with value units which can be checked? Hectare for example?

```
container Disc_alloc := 
	discrete_alloc(
		  LandUsePreparation/CBSKlasse/name		    	// 1 string 	attribute
		, Input/Compacted/ADomain				// 2 uint32	unit
		, Input/Compacted/Suitability				// 3 int32	container
		, const(0[Partitioning], LandUsePreparation/CBSKlasse)	// 4 uint8	attribute
		, Partitioning/name					// 5 string	attribute
		, Input/AggRegio					// 6 uint16	unit
		, Input/Compacted/AtomicRegionMap			// 7 uint16	attribute
		, Input/Claims/minclaims				// 8 uint32	container
		, Input/Claims/maxclaims				// 9 uint32	container
		, threshold						// 10 int32	parameter
		, FeasibleSolution					// 11 (empty)	container
	);
```

[Function-discrete-alloc](https://github.com/ObjectVision/GeoDMS/wiki/Function-discrete-alloc)

**Comment by IrenaItova** [comment]
Hi @jipclaassens , where is the full allocation function, I want to see it. It is not on the public repo, why is that?

**Reply by jipclaassens** [reply]
Which repo do you mean? The land use modelling one with the lus-demo?

**Reply by IrenaItova** [reply]
No, I mean the discrete_alloc (container ‘Disc_alloc’) function itself, the full version.

**Reply by jipclaassens** [reply]
Do you mean this? https://github.com/ObjectVision/GeoDMS/wiki/Allocation-functions

**Reply by IrenaItova** [reply]
I have seen this, that’s not it. I am looking for the actual function— the calculation rules

**Reply by jipclaassens** [reply]
Like the algorithm? @MaartenHilferink, maybe you can help with this?

**Reply by jipclaassens** [reply]
The third option on the page I sent has the syntax for the full version, which is further explained here: https://github.com/ObjectVision/GeoDMS/wiki/Function-discrete-alloc

**Reply by IrenaItova** [reply]
I am looking for the actual algorithm, it it's executable form. If that is in C++, it's also fine. The link you sheard has some examples of how you organize the attributes inside the function. But what I want to see is the script of the algorithm.

**Reply by IrenaItova** [reply]
I saw now Maarten shared it, no worries. Thanks!

**Comment by MaartenHilferink** [comment]
We have added simpler versions for discrete_alloc: 
- discrete_alloc_np without any partitioning
- discrete_alloc_sp with a single partitioning
in https://github.com/ObjectVision/GeoDMS/releases/tag/v16.0.3

**Comment by MaartenHilferink** [comment]
All discrete_alloc versions are implemented in https://github.com/ObjectVision/GeoDMS/blob/v16/geo/dll/src/DiscrAlloc.cpp
